app/models/guide.rb

```diff
@@ -13,4 +13,20 @@ class Guide < ActiveRecord::Base
   before_validation on: :create do |object|
     object.content_id = SecureRandom.uuid
   end
+
+  def work_in_progress_edition?
+    latest_edition.try(:state) == 'draft'
+  end
+
+  def update_attributes_from_params(params, state:)
```

Comment:
This makes it a required named parameter: if the client omits `state:`, an `ArgumentError: missing keyword: state` error will be raised, which in my opinion is more descriptive than `ArgumentError: wrong number of arguments`

I could have made them both named, but hesitated thinking this would appear confusing as the first parameter is a hash. What's your thinking?